### ISSUE TITLE
Cart Rules with specific action conditions doesn't work on the configurable (or composite?) products

### ISSUE THREAD

**Issue by sergei-sss** [magento/magento2#18366]
### Preconditions

1. Magento 2.2.0  with sample data

### Steps to reproduce
1. Disable all sample data cart rules.
2. Create a sales rule with action condition: 
```
If ALL  of these conditions are TRUE :
Material  contains  Polyester  
Color  is  Blue  
```
[cart rule edit page](https://drive.google.com/a/mgworx.com/file/d/13EH2C78mTLlHjriIqN28LWOwT7g0Tb4G/view?usp=drivesdk)

3. Go to category Men->Tops and add filter for Material -> Polyester
4. Select from the filtered products "Cassius Sparring Tank".
5. On the product page in the Additional Attributes : Material = LumaTech™, **Polyester**
6. Add to cart this product with Blue color and any size.

### Expected result
1. On the cart page must be discount from the cart rule.

### Actual result

1. Nothing, discount wasn't applied.

### Additional
This rule will work with any one condition:
- Material  contains  Polyester   (without color)
- Color is Blue  (without Material)

I suppose, cart item with configurable product doesn't contain color and cart item with simple product doesn't contain material...

**Comment by magento-engcom-team** [comment]
Hi @sergei-sss. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@sergei-sss do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by neodrg** [comment]
I think this is the same of #18305 Please test to verify this is not the same issue.

**Comment by sergei-sss** [comment]
@neodrg There are *cart* rules here.

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-andrii. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @sergei-sss . Thanks for the report. This problem has already been fixed on 2.2 develop and 2.3 develop.
![cassius](https://user-images.githubusercontent.com/41946904/48913308-aa964700-ee80-11e8-9347-dc18f0dae346.png)
